### DISCUSSION TITLE
Redirect to login page after accessing link to shared page

### DISCUSSION THREAD

**Discussion by ppa-xyt** [outline/outline#2650] [Ideas]
When not authenticated and using a link to a shared page, there shows an information that the page was not found:
![obraz](https://user-images.githubusercontent.com/48119126/136517947-5a9c929f-a557-4e7d-b599-f31cc629f61f.png)

After clicking `homepage` I'm redirected to Outline login page. After successful authentication I'm not redirected to the shared page url but to main page (dashboard).

I think it's not very user friendly as user has to use the link again instead of being redirected directly.

**Comment by tommoor** [comment]
This feels like typical behavior as there isn't a way to know in the client if redirecting back to the original page would result in another 404? Which would be a really bad experience post-login

**Comment by ppa-xyt** [comment]
I'm not sure if I explained clearly what I meant.

1. User goes to the shared (existing) page link: `https://outline.info/shared/abc`
2. "Not found 404" is shown with a link to homepage (let's assume it's OK)
3. User logs in
4. The page displayed after logging in is not a previously wanted `https://outline.info/shared/abc` but something other (last accessed page?) (not OK in my opinion)

I think most apps show the previously requested page (`https://outline.info/shared/abc`) after successful authentication.

I think there is a difference in behaviour between shared and standard pages. If unauthenticated user goes to shared page, "Not found" appeares (assuming the shared page link is valid). When unauthenticated user goes to standard (not shared) page link he/she is redirected to login page.

However in both cases user has to use the link again after authenticating (see point 4).

To sum up: 
1. Why 404 is shown for unauthenticated user even if the shared page link is valid? (point 2) - there should be a redirect to the login page if the page exists
2. Why after logging in the original page isn't displayed? (point 4) - the previously selected page should be displayed without any additional actions on user side
